homeassistant/components/utility_meter/sensor.py

```diff
@@ -147,7 +149,44 @@ async def async_setup_platform(
     )
 
 
-class UtilityMeterSensor(RestoreEntity, SensorEntity):
+@dataclass
+class UtilitySensorExtraStoredData(SensorExtraStoredData):
+    """Object to hold extra stored data."""
+
+    native_value: Decimal  # type: ignore[assignment]
```

Comment:
This is not a clean subclass of `SensorExtraStoredData` since we now fight the typing.
I'd suggest to add support for `Decimal` to `SensorExtraStoredData` directly instead so it can be used by other integrations too.
Or, add a new member `decimal_state` or similar here and store the decimal in it.

Reply:
I'll work a similar solution for the filter integration, so I have gone with your 1st proposal (added support in SensorExtraStoredData for Decimal)